sregistry/main/base/inspect.py

```diff
@@ -54,10 +54,14 @@ def get_metadata(self, image_file, names=None):
     if os.path.exists(singularity) and image_file:
         from spython.main import Client as Singularity
 
+        # Store the original quiet setting
+        is_quiet = Singularity.quiet
+
         # We try and inspect, but not required (wont work within Docker)
         try:
             Singularity.quiet = True
             updates = Singularity.inspect(image=image_file)
+            Singularity.quiet = is_quiet
```

Comment:
Would it make sense to restore the attribute value in a finally clause?

Reply:
Yes, good call. If the update fails we never hit the last statement (but we'll hit the first!). Change coming right up!